dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1002 @@
+    compute_api_client = CreateService('compute', 'beta')
+
+    request = compute_api_client.instances().aggregatedList(project=project_id)  # pylint: disable=no-member
+
+    while request is not None:
+      response = request.execute()
+
+      for zone in response['items'].values():
+
+        for instance in zone.get('instances', {}):
+          resource = Resource()
+          resource.id = instance.get('id')
+          resource.resource_name = instance.get('selfLink')
+          resource.creation_timestamp = instance.get('creationTimestamp')
+
+          for disk in instance.get('disks', {}):
+            disk_resource = Resource()
+            # 'source' here is just the full resouce name of the disk. It should
+            # be present for all disks
+            if disk.get('source'):
+              disk_resource.resource_name = disk.get('source')
+              # the disk ID is not present so we have to search for it in the
+              # list of processed resources
+              matched_disk = self._FindResource(disk_resource.name,
+                                                disk_resource.type,
+                                                disk_resource.zone,
+                                                disk_resource.project_id)
+              if matched_disk:
+                disk_resource = matched_disk
+
+              # Adding the instance as a child to the disk
+              disk_resource.children.add(resource)
+
+              # If the instance has a sourceMachineImage set, it means that this
+              # instance is created from a Machine Image. So we add this as a
+              # parent to the disk
+              if (instance.get('sourceMachineImage')
+                  and not disk_resource.parent):
+                disk_resource.parent = Resource()
+                disk_resource.parent.resource_name = instance.get(
+                    'sourceMachineImage')
+
+              # If the disk resource name is the same name as the instance name
+              # then this disk is the one created automatically with the
+              # incident so we set it as its parent. There is a exceptional case
+              # here where if a disk with the same name as the instance already
+              # created in the same zone, GCP will alter the default name. This
+              # is not handled here yet.
+              if disk_resource.name == resource.name:
+                if not resource.parent:
+                  resource.parent = disk_resource
+
+              if disk_resource.id:
+                self.resources_dict[disk_resource.id] = disk_resource
+
+            resource.disks.append(disk_resource)
+
+          result[resource.id] = resource
+
+      # pylint: disable=no-member
+      request = compute_api_client.instances().aggregatedList_next(
+          previous_request=request, previous_response=response)
+
+    return result
+
+  def _RetrieveListOfInstanceTemplates(self,
```

Comment:
same as _RetrieveListOfDiskImages

Reply:
libcloudforensics doesn't have the ability to list Instance Templates